ACE/ace/config-android.h

```diff
@@ -10,74 +10,66 @@
 // There is a large number of combinations of these two that can lead to
 // problems.
 
-// Location of the __ANDROID_API__ define
-// #include $NDK_ROOT/sysroot/usr/include/android/api-level.h
-#include "android/api-level.h"
-
-#if !defined (__ANDROID_API__)
-# error __ANDROID_API__ must be defined
-#endif
+#include <android/ndk-version.h>
+#include <android/api-level.h>
 
 #define ACE_ANDROID
 #define ACE_PLATFORM_CONFIG config-android.h
 
-#include "ace/config-linux-common.h"
+#include "config-linux-common.h"
 
 /*
  * Android NDK Revision Macros
  *
- * Revsions Scheme Work Like This:
- * Revision | __NDK_MAJOR__ | __NDK_MINOR__
- * r16      | 16            | 0
- * r16b     | 16            | 1
- * r16c     | 16            | 2
+ * Revisions Scheme Work Like This:
+ * Revision | __NDK_MAJOR__ | __NDK_MINOR__ | __NDK__BETA__
+ * r16      | 16            | 0             | 0
+ * r16b     | 16            | 1             | 0
+ * r16c     | 16            | 2             | 0
+ * r22-beta1| 22            | 0             | 1
+ *
+ * __NDK_BETA__ is starts at 1 and increments until release when it is 0.
  *
  * After r16, NDK version macros are defined in android/ndk-version.h Before
  * that they must be defined in platform_macros.GNU before the include of
  * platform_android.GNU.
  */
-#define ACE_ANDROID_NDK_AT_LEAST(MAJ, MIN) \
-  (__NDK_MAJOR__ > (MAJ) || (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ >= (MIN)))
-
-#define ACE_ANDROID_NDK_EXACTLY(MAJ, MIN) \
-  (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ == (MIN))
+#define ACE_ANDROID_NDK_AT_LEAST(MAJ, MIN, BET) (\
+  (__NDK_MAJOR__ > (MAJ)) || \
+  (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ >= (MIN)) || \
+  (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ == (MIN) && \
+    (__NDK_BETA__ == 0 || __NDK_BETA__ >= (BET)) \
+  ) \
+)
 
-#define ACE_ANDROID_NDK_LESS_THAN(MAJ, MIN) \
-  (__NDK_MAJOR__ < (MAJ) || (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ < (MIN)))
+#define ACE_ANDROID_NDK_EXACTLY(MAJ, MIN, BET) \
+  (__NDK_MAJOR__ == (MAJ) && __NDK_MINOR__ == (MIN) && __NDK_BETA__ == (BETA))
 
-#ifdef ACE_ANDROID_NDK_HAS_NDK_VERSION_H
-#  include "android/ndk-version.h"
-#else
-#  ifndef __NDK_MAJOR__
-#    error ndk-version.h is missing, __NDK_MAJOR__ for Android NDK must be defined!
-#  endif
-#  ifndef __NDK_MINOR__
-#    error ndk-version.h is missing, __NDK_MINOR__ for Android NDK must be defined!
-#  endif
-#endif
+#define ACE_ANDROID_NDK_LESS_THAN(MAJ, MIN, BET) \
+  !ACE_ANDROID_NDK_AT_LEAST((MAJ), (MIN), (BET))
 
 // ucontext.h and clock_settime() were added in r10c
-#if ACE_ANDROID_NDK_AT_LEAST(10, 2)
+#if ACE_ANDROID_NDK_AT_LEAST(10, 2, 0)
 #  define ACE_HAS_UCONTEXT_T
```

Comment:
Some cleanup possible now that r18 is our minimum?

Reply:
Yes, I was going to do that, but it slipped my mind.